### PR TITLE
Add Incentives Predeploy

### PR DESCRIPTION
**Pull request by ferrell-code**
companion to: https://github.com/AcalaNetwork/Acala/pull/2192

### CODE REVIEW

File: contracts/incentives/Incentives.sol [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+     * @dev Stake LP token to add shares to PoolId::Dex
+     * Returns a boolean value indicating whether the operation succeeded.
+     */
+    function depositDexShare(address currencyId, uint128 amount)
+    public
+    override
+    returns (bool) {
+        require(amount != 0, "Incentives: amount is zero");
+
+        (bool success, bytes memory returnData) = precompile.staticcall(abi.encodeWithSignature("depositDexShare(address,address,uint128)", msg.sender, currencyId, amount));
```
Comment (zjb0807) [inline contracts/incentives/Incentives.sol:58]:
```suggestion
        (bool success, bytes memory returnData) = precompile.call(abi.encodeWithSignature("depositDexShare(address,address,uint128)", msg.sender, currencyId, amount));

```

File: contracts/incentives/Incentives.sol [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+     * Returns a boolean value indicating whether the operation succeeded.
+     */
+    function withdrawDexShare(address currencyId, uint128 amount)
+    public
+    override
+    returns (bool) {
+        require(amount != 0, "Incentives: amount is zero");
+
+        (bool success, bytes memory returnData) = precompile.staticcall(abi.encodeWithSignature("withdrawDexShare(address,address,uint128)", msg.sender, currencyId, amount));
+        assembly {
```
Comment (zjb0807) [inline contracts/incentives/Incentives.sol:80]: same here

File: contracts/incentives/Incentives.sol [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    /**
+     * @dev Claim all avalible multi currencies rewards for specific PoolId
+     * Returns a boolean value indicating whether the operation succeeded.
+     */
+    function claimRewards(PoolId pool, address poolCurrencyId)
+    public
+    override
+    returns (bool) {
+        (bool success, bytes memory returnData) = precompile.staticcall(abi.encodeWithSignature("claimRewards(address,PoolId,address)", msg.sender, pool, poolCurrencyId));
```
Comment (zjb0807) [inline contracts/incentives/Incentives.sol:98]: same here

File: contracts/incentives/Incentives.sol (hunk 1 above)
Comment (ferrell-code) [reply contracts/incentives/Incentives.sol]: Good catch!

**Review by zjb0807 — APPROVED** [review]
LGTM

**Pull request merged**